### PR TITLE
chore: add script to help automate discovery document updating

### PR DESCRIPTION
**Pull request by scotthart**
part of the work for #11737

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13733)

### CODE REVIEW

File: generator/discovery/update_discovery_doc.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+source "$(dirname "$0")/../../ci/lib/init.sh"
+source module ci/lib/io.sh
+
+if (($# > 0)); then
+  cat <<EOM
```
Comment (devbww) [inline generator/discovery/update_discovery_doc.sh:23]: `1>&2`

File: generator/discovery/update_discovery_doc.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  JSON.
+EOM
+  exit 1
+fi
+
+readonly COMPUTE_DISCOVERY_DOCUMENT_URL="https://www.googleapis.com/discovery/v1/apis/compute/v1/rest"
+readonly COMPUTE_DISCOVERY_JSON_RELATIVE_PATH="generator/discovery/compute_public_google_rest_v1.json"
+
+io::log_h2 "Fetching discovery document from ${COMPUTE_DISCOVERY_DOCUMENT_URL}"
+curl "${COMPUTE_DISCOVERY_DOCUMENT_URL}" >"${PROJECT_ROOT}"/"${COMPUTE_DISCOVERY_JSON_RELATIVE_PATH}"
```
Comment (devbww) [inline generator/discovery/update_discovery_doc.sh:36]: Might as well make the output file a single quoted string, like the same path a few lines down.

File: ci/cloudbuild/builds/generate-libraries.sh [hunk 3]
```diff
@@ -72,6 +77,12 @@ if [ -z "${GENERATE_GOLDEN_ONLY}" ]; then
     --output_path="${PROJECT_ROOT}" \
     --check_parameter_comment_substitutions=true \
     --config_file="${PROJECT_ROOT}/generator/generator_config.textproto"
+
+  if [ -n "${UPDATED_DISCOVERY_DOCUMENT}" ]; then
+    io::log_h2 "Adding new ${UPDATED_DISCOVERY_DOCUMENT} generated service code"
+    git add "${PROJECT_ROOT}/google/cloud/${UPDATED_DISCOVERY_DOCUMENT}"
+  fi
+
```
Comment (devbww) [inline ci/cloudbuild/builds/generate-libraries.sh:85]: Nit: A blank line before an `else` doesn't match our style.

**Review by devbww — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13733)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @devbww)

___
*[`ci/cloudbuild/builds/generate-libraries.sh` line 85 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13733#gh-1513782470:-NsJoQc6BUdBNIvYicQA:bss4a2b) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/f9c6991a9589598620f27ae7caa6fb40e5606dc3/ci/cloudbuild/builds/generate-libraries.sh#L85)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Nit: A blank line before an `else` doesn't match our style.
</blockquote></details>

Removed.
___
*[`generator/discovery/update_discovery_doc.sh` line 23 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13733#gh-1513779812:-NsJoS8s7Di7oApSGqex:bqmjwcz) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/f9c6991a9589598620f27ae7caa6fb40e5606dc3/generator/discovery/update_discovery_doc.sh#L23)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

`1>&2`
</blockquote></details>

Added.
___
*[`generator/discovery/update_discovery_doc.sh` line 36 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13733#gh-1513780355:-NsJoT5JAm9QB0szcMtE:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/f9c6991a9589598620f27ae7caa6fb40e5606dc3/generator/discovery/update_discovery_doc.sh#L36)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Might as well make the output file a single quoted string, like the same path a few lines down.
</blockquote></details>

Done.

**Pull request merged**